farm/modeling/optimization.py

```diff
@@ -263,8 +265,8 @@ def get_scheduler(optimizer, opts):
 
 def calculate_optimization_steps(n_batches, grad_acc_steps, n_epochs, local_rank):
     optimization_steps = int(n_batches / grad_acc_steps) * n_epochs
-    if local_rank != -1:
-        optimization_steps = optimization_steps // torch.distributed.get_world_size()
+    # if local_rank != -1:
```

Comment:
Do we need this?

Reply:
No, we can get rid of it since `n_batches` is already taking into account the number of batches for the current rank.